src/SDWaveFile.cpp

```diff
@@ -224,10 +259,30 @@ void SDWaveFile::readHeader()
     _file.close();
     return;
   }
-
   struct WaveFileHeader header;
 
-  if (_file.read(&header, sizeof(header)) != sizeof(header)) {
+  headerSize = sizeof(struct WaveFileHeader) - sizeof(header.subChunk2);
+  //Read the first WAV header's chunk
+  if (_file.read(&header, headerSize) != headerSize) {
+    _file.close();
+    return;
+  }
+
+  if (header.subChunk1.audioFormat != 1) {
+    _file.close();
+    return;
+  }
+
+  _headerOffset = findData(headerSize);
+  if (_headerOffset == -1) {
+    _file.close();
+    return;
+  }
+  //if findData goes fine, data label is assigned to subChunk2.id
+  header.subChunk2.id = 0x64617461;
```

Comment:
This seems strange to assign this here, we can just remove the check below instead?